internal/eventlog/eventlog.go

```diff
@@ -0,0 +1,47 @@
+package eventlog
+
+import (
+	"fmt"
+	"time"
+)
+
+type Log struct {
+	Events []Event
+}
+
+func (l *Log) Append(e *Event) {
+	if e != nil {
+		l.Events = append(l.Events, *e)
+	}
+}
+
+type Event struct {
+	Success     bool
+	Name        string
+	Message     string
+	StartedAt   time.Time
+	CompletedAt time.Time
+}
```

Comment:
Since this needs to be replicated in enduro, what are you expecting to find in the `Events[]` passed in the workflow result? My concern is that we are passing events only to create preservation tasks in enduro right now, so it may be better to call them `PreservationTaskEvents`, or just `PreservationTasks`. If you are envisioning this slice to contain other types of events, should it include a type field in the payload or find another way to differentiate even types in enduro?

I think that may be hard to maintain from both sides, so I'd rather call this slice `PreservationTasks` and have another one(s) in the future if we need to pass other types of events.

Reply:
Okay, to get into the pedantic semantics a bit here: _is_ SIP validation a preservation activity or is it a pre-ingest activity? :stuck_out_tongue_winking_eye: 

The `Event` struct doesn't map one-to-one to a [PreservationTask](https://github.com/artefactual-sdps/enduro/blob/main/internal/datatypes/preservation_task.go#L11), so I don't think it's necessary for them to use the same name.  I did intend for the `eventlog` package to be a fairly general event logging module, so it could be reused in other contexts (via copy/paste as it's an internal package).  I'm fine with adding a `Event.Type` field later if we need to distinguish an Event that will not become a PreservationTask, but that doesn't seem useful to me before we have distinct event types.

I think you raise a good point about defining the Event type here and in Enduro though - because Event is really defining an interface to Enduro, I think it's better to define the type in Enduro if possible, and then import it into preprocessing.  Alternatively we could create a protocol buffer definition of for Events (or PreservationTasks, or whatever we call it) and use that for communicating between Preprocessing and Enduro, though this seem like more work.

Reply:
I also just realized that these same events are going to be serialized as PREMIS events at some point, so maybe these events should be structured as PREMIS events?  PREMIS is pretty complex though and I don't want to complicate these events unless it's necessary. :thinking:

Reply:
> I think you raise a good point about defining the Event type here and in Enduro though - because Event is really defining an interface to Enduro, I think it's better to define the type in Enduro if possible, and then import it into preprocessing. Alternatively we could create a protocol buffer definition of for Events (or PreservationTasks, or whatever we call it) and use that for communicating between Preprocessing and Enduro, but this seem like more work.

Nevermind, importing the Event type from Enduro is probably a bad idea - the Event definition for this project should be owned by this project so external changes don't break the local implementation.

Reply:
Don't worry too much about the naming or if they match one to one. My take is that an event type is hard to maintain across multiple repositories, and that re-using the same payload for different events may be more prone to backward incompatible changes.